src/mango.ts

```diff
@@ -0,0 +1,491 @@
+    console.log(scalper.symbol, 'Delta Netural: Slippage', deltaThreshold);
+    return;
+  }
+  console.log(scalper.symbol, 'Outside delta threshold:', Math.abs(hedgeDeltaTotal), 'vs.', deltaThreshold);
+
+  // Determine spot or perp order based on funding rate.
+  const bidSide = await perpMarket.loadBids(mangoClient);
+  const askSide = await perpMarket.loadAsks(mangoClient);
+  const fundingRate = (24 * 365)
+    * perpMarket.getCurrentFundingRate(bidSide, askSide);
+  const buySpot = fundingRate > PERP_FUNDING_RATE_THRESHOLD;
+  const sellSpot = -fundingRate < PERP_FUNDING_RATE_THRESHOLD;
+  let hedgeProduct = getMangoHedgeProduct(hedgeSide, buySpot, sellSpot);
+  // No OpenBook BTC market exists still
+  if (scalper.symbol === 'BTC') {
+    hedgeProduct = HedgeProduct.Perp;
+  }
+
+  console.log(
+    `${scalper.symbol} Target Delta Hedge: ${hedgeSideText} ${hedgeProduct} ${-hedgeDeltaTotal} \
+      DIP Δ: ${dipTotalDelta} Mango Perp Δ: ${mangoPerpDelta} Mango Spot Δ: ${mangoSpotDelta} \
+      Spot Δ: ${spotDelta} Offset Δ ${scalper.deltaOffset} Fair Value: ${fairValue}`,
+  );
+
+  // Determine what price to use for hedging depending on allowable slippage.
+  hedgePrice = roundPriceToTickSize(hedgeDeltaTotal < 0
+    ? fairValue * (1 + slippageTolerance) : fairValue * (1 - slippageTolerance), scalper.tickSize);
+
+  // Break up order depending on whether the book can support it
+  const bookSide = hedgeDeltaTotal < 0 ? askSide : bidSide;
+  const hedgeDeltaClip = roundQtyToSpotSize(hedgeDeltaTotal
+      / orderSpliceMango(
+        hedgeDeltaTotal,
+        fairValue,
+        maxNotional.get(scalper.symbol),
+        slippageTolerance,
+        bookSide,
+        perpMarket,
+      ), scalper.minSize);
+
+  const deltaOrderId = new Date().getTime() * 2;
+
+  // Start listening for Delta Hedge Fills
+  const deltaFillListener = async (event: WebSocket.MessageEvent) => {
+    const parsedEvent = JSON.parse(event.data as string);
+    const {
+      owner, clientOrderId, quantity, price,
+    } = parsedEvent.event;
+    if (owner === mangoAccount.publicKey.toBase58() && clientOrderId === deltaOrderId) {
+      const fillQty = (hedgeSide === PerpOrderSide.bid ? 1 : -1) * quantity;
+      hedgeDeltaTotal += fillQty;
+      console.log(`${scalper.symbol} Delta Filled ${hedgeSideText} ${hedgeProduct} Qty ${fillQty} \
+          Price ${price} Remaining ${hedgeDeltaTotal} ID ${deltaOrderId} ${new Date().toUTCString()}`);
+    }
+  };
+
+  // Setup a listener for the order.
+  if (fillFeed.readyState === WebSocket.OPEN) {
+    fillFeed.addEventListener('message', deltaFillListener);
+    console.log(scalper.symbol, 'Listening For Delta Hedges');
+  } else {
+    console.log(scalper.symbol, 'Websocket State', fillFeed.readyState);
+  }
+
+  console.log(`${scalper.symbol} ${hedgeSideText} ${hedgeProduct} ${Math.abs(hedgeDeltaClip)} \
+    Limit: ${hedgePrice} # ${deltaHedgeCount} ID ${deltaOrderId}`);
+
+  try {
+    // TODO: Test Spot
```

Comment:
Has this not been tested? Should either test it or remove it.
And ideally there is just unit test coverage, but would probably want to go back and continue the break up and clean up we were doing before first

Reply:
Tested for wBTCpo now